dlls/winevulkan/vulkan.c

```diff
@@ -2473,11 +2211,20 @@ NTSTATUS wine_vkCreateSwapchainKHR(void *args)
         native_info.imageExtent = object->real_extent;
         native_info.imageUsage |= VK_IMAGE_USAGE_TRANSFER_DST_BIT; /* XXX: check if supported by surface */
 
+        if (caps.supportedUsageFlags & VK_IMAGE_USAGE_STORAGE_BIT){
+            native_info.imageUsage |= VK_IMAGE_USAGE_STORAGE_BIT;
+        }
+
         if(native_info.imageFormat != VK_FORMAT_B8G8R8A8_UNORM &&
                 native_info.imageFormat != VK_FORMAT_B8G8R8A8_SRGB){
             FIXME("swapchain image format is not BGRA8 UNORM/SRGB. Things may go badly. %d\n", native_info.imageFormat);
         }
 
+        device->phys_dev->instance->funcs.p_vkGetPhysicalDeviceFormatProperties(device->phys_dev->phys_dev, native_info.imageFormat, &formatProperties);
+        if(!(formatProperties.optimalTilingFeatures & VK_FORMAT_FEATURE_STORAGE_IMAGE_BIT)){
+            native_info.imageFormat = VK_FORMAT_B8G8R8A8_UNORM;
```

Comment:
This breaks SRGB swapchains if the application uses it as a vkCmdBlitImage destination. It will also cause validation errors for image views that the application creates, but that fine I guess because no driver actually cares afaik.